### PR TITLE
Refactor eth crypto

### PR DESCRIPTION
**Pull request by nickeskov**
Also added test when ethereum senderPK starts with leading zero(es).

### CODE REVIEW

File: pkg/proto/eth_crypto.go [hunk 1]
```diff
@@ -52,6 +52,19 @@ type EthereumSignature struct {
 	sig [ethereumSignatureLength]byte
 }
 
+func NewEthereumSignatureFromVRS(v byte, R, S *big.Int) (EthereumSignature, error) {
```
Comment (alexeykiselev) [inline pkg/proto/eth_crypto.go:55]: Bad naming for `R` and `S`. Please, make them lowercase.

File: pkg/proto/eth_signer.go [hunk 2]
```diff
@@ -192,7 +190,7 @@ func (s londonSigner) SenderPK(tx *EthereumTransaction) (*EthereumPublicKey, err
 	if tx.ChainId().Cmp(s.chainId) != 0 {
 		return nil, ErrInvalidChainId
 	}
-	return recoverEthereumPubKey(s.Hash(tx), R, S, V, true)
+	return recoverEthereumPubKey(s.Hash(tx), R, S, V)
```
Comment (alexeykiselev) [inline pkg/proto/eth_signer.go:193]: Please, change to lowercase everywhere.

File: pkg/proto/eth_signer.go [hunk 3]
```diff
@@ -511,24 +509,18 @@ func decodeSignature(sig []byte, legacyV bool) (r, s, v *big.Int, err error) {
 	return r, s, v, nil
 }
 
-func recoverEthereumPubKey(sighash EthereumHash, R, S, Vb *big.Int, homestead bool) (*EthereumPublicKey, error) {
-	if Vb.BitLen() > 8 {
+func recoverEthereumPubKey(sighash EthereumHash, R, S, V *big.Int) (*EthereumPublicKey, error) {
```
Comment (alexeykiselev) [inline pkg/proto/eth_signer.go:512]: Especially here

**Review by alexeykiselev — CHANGES_REQUESTED** [review]

File: pkg/proto/eth_crypto.go (hunk 1 above)
Comment (nickeskov) [reply pkg/proto/eth_crypto.go]: Fixed

File: pkg/proto/eth_signer.go (hunk 2 above)
Comment (nickeskov) [reply pkg/proto/eth_signer.go]: Changed

**Review by alexeykiselev — APPROVED** [review]
